K2Field.K2NE.ServiceBroker/ServiceObjects/ExcelServices/ExcelImportServiceSO.cs

```diff
@@ -0,0 +1,528 @@
+
+                                    // handle Date amd DateTime columns correctly.  Convert Excel datetime format (double) to .NET DateTime type
+                                    if (nonblankCharCount > 0 && newSmartObject.Properties[sColName].Type == SOC.PropertyType.DateTime) // DateTime column
+                                    {
+                                        tmpDate = DateTime.FromOADate(Convert.ToDouble(dr[sColName].ToString()));
+                                        newSmartObject.Properties[sColName].Value = String.Concat(tmpDate.ToShortDateString(), " ", tmpDate.ToShortTimeString());
+                                    }
+                                    else if (nonblankCharCount > 0 && newSmartObject.Properties[sColName].Type == SOC.PropertyType.Date) // Date column
+                                    {
+                                        tmpDate = DateTime.FromOADate(Convert.ToDouble(dr[sColName].ToString()));
+                                        newSmartObject.Properties[sColName].Value = tmpDate.ToShortDateString();
+                                    }
+                                    else if (nonblankCharCount > 0 && newSmartObject.Properties[sColName].Type == SOC.PropertyType.Time) // Time column
+                                    {
+                                        tmpDate = DateTime.FromOADate(Convert.ToDouble(dr[sColName].ToString()));
+                                        newSmartObject.Properties[sColName].Value = tmpDate.ToShortTimeString();
+                                    }
+                                    else // not a Date or DateTime column
+                                    {
+                                        newSmartObject.Properties[sColName].Value = dr[sColName].ToString();
+                                    }
+                                }
+
+                                // Add the transaction ID value for identification purposes.
+                                if (sTransactionIDName != string.Empty && sTransactionIDValue != string.Empty)
+                                {
+                                    // replace spaces with underscores as SmartObject system names do that
+                                    newSmartObject.Properties[sTransactionIDName.Replace(" ", "_")].Value = sTransactionIDValue;
+                                }
+
+                                if (!string.IsNullOrEmpty(FQN))
+                                {
+                                    newSmartObject.Properties["FQN"].Value = FQN;
+                                }
+
+                                inputList.SmartObjectsList.Add(newSmartObject);
+                            }
+                            try
+                            {
+                                smoClientServer.ExecuteBulkScalar(soImport, inputList);
+                            }
+                            catch (SOC.SmartObjectException ex)
```

Comment:
The exception handling in the Execute() method already does something similar. See also next comment.

Reply:
Removed.